hikari/impl/entity_factory.py

```diff
@@ -767,61 +767,81 @@ def _deserialize_member_move_entry_info(
-            if (raw_user_id := entry_payload["user_id"]) is not None:
-                user_id = snowflakes.Snowflake(raw_user_id)
+        target_id: typing.Optional[snowflakes.Snowflake] = None
+        if (raw_target_id := payload["target_id"]) is not None:
+            target_id = snowflakes.Snowflake(raw_target_id)
 
-            action_type: typing.Union[audit_log_models.AuditLogEventType, int]
-            action_type = audit_log_models.AuditLogEventType(entry_payload["action_type"])
+        user_id: typing.Optional[snowflakes.Snowflake] = None
+        if (raw_user_id := payload["user_id"]) is not None:
+            user_id = snowflakes.Snowflake(raw_user_id)
 
-            options: typing.Optional[audit_log_models.BaseAuditLogEntryInfo] = None
-            if (raw_option := entry_payload.get("options")) is not None:
-                if option_converter := self._audit_log_event_mapping.get(action_type):
-                    options = option_converter(raw_option)
+        action_type: typing.Union[audit_log_models.AuditLogEventType, int]
+        action_type = audit_log_models.AuditLogEventType(payload["action_type"])
 
-                else:
-                    _LOGGER.debug("Unknown audit log action type found %r", action_type)
-                    continue
+        options: typing.Optional[audit_log_models.BaseAuditLogEntryInfo] = None
+        if (raw_option := payload.get("options")) is not None:
+            if option_converter := self._audit_log_event_mapping.get(action_type):
+                options = option_converter(raw_option)
 
-            entries[entry_id] = audit_log_models.AuditLogEntry(
-                app=self._app,
-                id=entry_id,
-                target_id=target_id,
-                changes=changes,
-                user_id=user_id,
-                action_type=action_type,
-                options=options,
-                reason=entry_payload.get("reason"),
-            )
+            else:
+                raise errors.UnrecognisedEntityError(f"Unknown audit log action type found {action_type!r}")
+
+        return audit_log_models.AuditLogEntry(
+            app=self._app,
+            id=entry_id,
+            target_id=target_id,
+            changes=changes,
+            user_id=user_id,
+            action_type=action_type,
+            options=options,
+            reason=payload.get("reason"),
+            guild_id=guild_id,
+        )
+
+    def deserialize_audit_log(
+        self, payload: data_binding.JSONObject, *, guild_id: snowflakes.Snowflake
+    ) -> audit_log_models.AuditLog:
+        entries = {}
+        for entry_payload in payload["audit_log_entries"]:
+            try:
+                entry = self.deserialize_audit_log_entry(entry_payload, guild_id=guild_id)
+
+            except errors.UnrecognisedEntityError as exc:
+                _LOGGER.debug(exc.reason)
+
+            else:
+                entries[entry.id] = entry
```

Comment:
```suggestion
                entries[entry.id] = entry
 
            except errors.UnrecognisedEntityError as exc:
                _LOGGER.debug(exc.reason)
```

Reply:
I prefer limiting the scope of the try block to the bit I actually expect to raise when possible, tends to help with readability

Reply:
What I dont like here is the lil jump that it does. In this case, the code isnt that big, and it should be quite clear that adding it to a dict shoudnt be raising a `UnrecognisedEntityError`

Reply:
that jump is just the natural flow of a try, except, else statement though